src/nrf_802154_types.h

```diff
@@ -138,6 +138,15 @@ typedef uint8_t nrf_802154_ack_data_t;
 #define NRF_802154_ACK_DATA_PENDING_BIT 0x00
 #define NRF_802154_ACK_DATA_IE          0x01
 
+/**
+ * @brief Methods of source address matching.
+ */
+typedef uint8_t nrf_802154_src_match_t;
+
+#define NRF_802154_SRC_MATCH_THREAD   0x00 // !< Implementation for Thread protocol.
+#define NRF_802154_SRC_MATCH_ZIGBEE   0x01 // !< Implementation for ZigBee protocol.
+#define NRF_802154_SRC_MATCH_STANDARD 0x02 // !< Standard-compliant implementation - PB is always set to 1.
```

Comment:
BTW, when you disable source matching mechanism, you get the same functionality, right? We don't have to provide multiple ways to achieve the same effect.

Reply:
You do, however I still think these should be separate. I may be mistaken here, as I don't know exact use cases, but it seems to me that the third matching method would be implemented so that the user is not restricted to thread or zigbee application, while disabling or enabling the entire auto pending bit mechanism could be done in run-time (at least I didn't find anything that would indicate otherwise).

Reply:
Let's discuss it F2F - should be faster :)